lib/pure/asyncdispatch.nim

```diff
@@ -1182,8 +1185,14 @@ else:
           else:
             retFuture.fail(newException(OSError, osErrorMsg(lastError)))
       else:
-        register(client.AsyncFD)
-        retFuture.complete((getAddrString(cast[ptr SockAddr](addr sockAddress)), client.AsyncFD))
+        try:
+          let address = getAddrString(cast[ptr SockAddr](addr sockAddress))
+          register(client.AsyncFD)
+          retFuture.complete((address, client.AsyncFD))
+        except:
+          # getAddrString may raise
```

Comment:
> Perhaps we should be using exception tracking to guarantee this?

That's a very good idea and the .async macro can inject a ``.raises: []`` to accomplish this (right?), but let's do it later, it will break even more code out there.